File: 14-LongestCommonPrefix/14-LongestCommonPrefix.py
# Last updated: 11/8/2025, 9:37:23 PM
class Solution(object):
    def longestCommonPrefix(self, strs):
        """
        :type strs: List[str]
        :rtype: str
        """
        word = strs[0]
        for i in range(1, len(strs)):
            new_word = ""
            for j in range(len(strs[i])):
                if j < len(word) and strs[i][j] == word[j]:
                    new_word += strs[i][j]
                else:
                    break
            word = new_word
        return word

File: 14-LongestCommonPrefix/14-LongestCommonPrefix.cpp
// Last updated: 11/8/2025, 9:36:41 PM
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        string word = strs[0];
        for (int i = 1; i < strs.size(); i++){
            string newWord = "";
            for (int j = 0; j < strs[i].length(); j++){
                if (strs[i][j] == word[j]){
                    newWord += strs[i][j];
                }
                else {
                    break;
                }
            }
            word = newWord;
        }
        return word;
    }
};